### PR TITLE
Support asynchronous generation

### PR DESCRIPTION
**Pull request by Richienb**
Fixes: #16 

// @sindresorhus

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -60,19 +60,36 @@ declare namespace cryptoRandomString {
...
+	//=> '2cf05d94db'
+	```
+	*/
+	(options?: cryptoRandomString.Options): string;
+
+	/**
+	Asynchronously generate a [cryptographically strong](https://en.wikipedia.org/wiki/Strong_cryptography) random string.
 
-@returns A randomized string.
+	@returns A randomized string.
```
Comment (sindresorhus) [inline index.d.ts:82]: Not correct

File: index.test-d.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ expectType<string>(cryptoRandomString({length: 10}));
 expectType<string>(cryptoRandomString({length: 10, type: 'url-safe'}));
 expectType<string>(cryptoRandomString({length: 10, type: 'numeric'}));
 expectType<string>(cryptoRandomString({length: 10, characters: '1234'}));
+expectType<Promise<string>>(cryptoRandomString.async({ length: 10 }));
```
Comment (sindresorhus) [inline index.test-d.ts:8]:
```suggestion
expectType<Promise<string>>(cryptoRandomString.async({length: 10}));
```

File: readme.md [hunk 3]
```diff
@@ -40,6 +40,10 @@ cryptoRandomString({length: 10, characters: 'abc'});
 
 Returns a randomized string. [Hex](https://en.wikipedia.org/wiki/Hexadecimal) by default.
 
+### cryptoRandomString.async(options)
+
+Returns promise which resolves with a randomized string. [Hex](https://en.wikipedia.org/wiki/Hexadecimal) by default.
```
Comment (sindresorhus) [inline readme.md:45]:
```suggestion
Returns a promise which resolves to a randomized string. [Hex](https://en.wikipedia.org/wiki/Hexadecimal) by default.
```

**Comment by Richienb** [comment]
@sindresorhus Ready for re-review!

File: index.d.ts [hunk 4]
```diff
@@ -60,19 +60,36 @@ declare namespace cryptoRandomString {
...
+	//=> '2cf05d94db'
+	```
+	*/
+	(options?: cryptoRandomString.Options): string;
+
+	/**
+	Asynchronously generate a [cryptographically strong](https://en.wikipedia.org/wiki/Strong_cryptography) random string.
 
-@returns A randomized string.
+	@returns A promise which resolves to randomized string.
```
Comment (sindresorhus) [inline index.d.ts:82]:
```suggestion
	@returns A promise which resolves to a randomized string.
```

**Comment by niftylettuce** [comment]
@sindresorhus can you please do another version release?  this isn't on npm yet

**Comment by Richienb** [comment]
@niftylettuce It will probably be released once https://github.com/sindresorhus/crypto-random-string/pull/25 is merged.

**Pull request merged**